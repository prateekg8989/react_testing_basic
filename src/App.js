import './App.css';
import { useState } from "react";

function App() {

  const [count, setCount] = useState(0);

  return (
    <div className="App">
      <p>Count: {count}</p>
      <button className="increment" type="button" onClick={() => setCount(count+1)}>Click me</button>
    </div>
  );
}

export default App;
