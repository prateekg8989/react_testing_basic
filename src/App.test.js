import { render, screen } from '@testing-library/react';
import App from './App';
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';


describe('Addition', () => {
  it('knows that 2 and 2 make 4', () => {
    expect(2 + 2).toBe(4);
  });
});

describe('App component', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
  })

  it('starts with a count of 0', () => {
    const text = wrapper.find('p').text();
    expect(text).toEqual('Count: 0');
  });

  it('increments count by 1 when the increment button is clicked', () => {
    wrapper.setProps({'test': 'value'});
    const incrementBtn = wrapper.find('button.increment');
    incrementBtn.simulate('click');
    const text = wrapper.find('p').text();
    expect(text).toEqual('Count: 1');
  });

  it('matches the snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
