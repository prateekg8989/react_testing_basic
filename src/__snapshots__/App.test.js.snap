// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches the snapshot 1`] = `
<div
  className="App"
>
  <p>
    Count: 
    0
  </p>
  <button
    className="increment"
    onClick={[Function]}
    type="button"
  >
    Click me
  </button>
</div>
`;
